quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs

```diff
@@ -765,37 +766,76 @@ impl Metastore for PostgresqlMetastore {
         index_id: &str,
         split_ids: &[&'a str],
     ) -> MetastoreResult<()> {
-        run_with_tx!(self.connection_pool, tx, {
-            let deletable_states = [
-                SplitState::Staged.as_str(),
-                SplitState::MarkedForDeletion.as_str(),
-            ];
-            let deleted_split_ids: Vec<String> = sqlx::query_scalar(
-                r#"
+        const DELETE_SPLITS_QUERY: &str = r#"
+            -- Select the splits to delete, regardless of their state.
+            -- The left join make it possible to identify the splits that do not exist.
+            WITH input_splits AS (
+                SELECT input_splits.split_id, splits.split_state
+                FROM UNNEST($2) AS input_splits(split_id)
+                LEFT JOIN (
+                    SELECT split_id, split_state
+                    FROM splits
+                    WHERE
+                        index_id = $1
+                        AND split_id = ANY($2)
+                    FOR UPDATE
+                    ) AS splits
+                USING (split_id)
+            ),
+            -- Delete the splits iff all the splits are marked for deletion.
```

Comment:
```suggestion
            -- Delete the splits if and only if all the splits are marked for deletion.
```

I love iff too, but several developers reading my comments thought it was a spelling mistake.